codec-http/src/main/java/io/netty/handler/codec/rtsp/RtspDecoder.java

```diff
@@ -0,0 +1,180 @@
+
+    /**
+     * Creates a new instance with the default
+     * {@code maxInitialLineLength (4096)}, {@code maxHeaderSize (8192)}, and
+     * {@code maxContentLength (8192)}.
+     */
+    public RtspDecoder() {
+        this(DEFAULT_MAX_INITIAL_LINE_LENGTH,
+             DEFAULT_MAX_HEADER_SIZE,
+             DEFAULT_MAX_CONTENT_LENGTH);
+    }
+
+    /**
+     * Creates a new instance with the specified parameters.
+     * @param maxInitialLineLength The max allowed length of initial line
+     * @param maxHeaderSize The max allowed size of header
+     * @param maxContentLength The max allowed content length
+     */
+    public RtspDecoder(final int maxInitialLineLength,
+                       final int maxHeaderSize,
+                       final int maxContentLength) {
+        super(maxInitialLineLength, maxHeaderSize, maxContentLength * 2, false);
+    }
+
+    /**
+     * Creates a new instance with the specified parameters.
+     * @param maxInitialLineLength The max allowed length of initial line
+     * @param maxHeaderSize The max allowed size of header
+     * @param maxContentLength The max allowed content length
+     * @param validateHeaders Set to true if headers should be validated
+     */
+    public RtspDecoder(final int maxInitialLineLength,
+                       final int maxHeaderSize,
+                       final int maxContentLength,
+                       final boolean validateHeaders) {
+        super(maxInitialLineLength,
+              maxHeaderSize,
+              maxContentLength * 2,
+              false,
+              validateHeaders);
+    }
+
+    @Override
+    protected final HttpMessage createMessage(final String[] initialLine)
+            throws Exception {
+        // If the first element of the initial line is a version string then
+        // this is a response
+        if (versionPattern.matcher(initialLine[0]).matches()) {
+            isDecodingRequest = false;
+            return new DefaultHttpResponse(RtspVersions.valueOf(initialLine[0]),
+                new HttpResponseStatus(Integer.parseInt(initialLine[1]),
+                                       initialLine[2]),
+                validateHeaders);
+        } else {
+            isDecodingRequest = true;
+            return new DefaultHttpRequest(RtspVersions.valueOf(initialLine[2]),
+                    RtspMethods.valueOf(initialLine[0]),
+                    initialLine[1],
+                    validateHeaders);
+        }
+    }
+
+    @Override
+    protected final boolean isContentAlwaysEmpty(final HttpMessage msg) {
+        // Unlike HTTP, RTSP always assumes zero-length body if Content-Length
+        // header is absent.
+        boolean empty = super.isContentAlwaysEmpty(msg);
+        if (empty) {
+            return true;
+        }
+        if (!msg.headers().contains(RtspHeaderNames.CONTENT_LENGTH)) {
+            return true;
+        }
+        return empty;
```

Comment:
I think this can be simplified as:

```java
    return super.isContentAlwaysEmpty(msg) || !msg.headers().contains(RtspHeaderNames.CONTENT_LENGTH);
```